bundle/regal/rules/style/trailing_default_rule.rego

```diff
@@ -0,0 +1,24 @@
+# METADATA
+# description: Default rule should be declared first
+package regal.rules.style["trailing-default-rule"]
+
+import rego.v1
+
+import data.regal.ast
+import data.regal.result
+
+report contains violation if {
+	some i, rule in input.rules
+
+	rule["default"] == true
+
+	name := ast.ref_to_string(rule.head.ref)
+	name in all_names(array.slice(input.rules, 0, i))
```

Comment:
🤔 how should I read this? if the default rule is in the 0-ith range of rules? Perhaps it's too early, but it's not clear to me how this is working 😅

Reply:
```
array.slice(input.rules, 0, i)
```
provides a slice of all the rules up until `i`, i.e the `default` rule.. and then we check if the same name of the default has been used for any of those.

Reply:
Ahh, yes. that does make sense. I'd assumed that the implementation was going to be like: make sure that the rule is the first when ordered by line or similar, but this works too! thanks.